app/index.js

```diff
@@ -267,20 +267,50 @@ server.route({
       let { query } = request;
       let targetYear = null;
 
-      // Check for redis data with this query.
+      // Generate cache key from query params
       const cacheKey = JSON.stringify({ id, query });
-      if (redisEnabled) {
-        const cachedData = await redis.get(cacheKey);
+      if (cache.enabled) {
+        // Load cache entry
+        let cachedData = await redisClient.getObject(cacheKey);
```

Comment:
There's still a race condition here, in that we're looking for a result, and if we don't get it, we're writing back. We can fix that race condition at the cost of requiring 2 hits to redis for each request:
```
// do exclusive set to do cheap lock
result = setnx key val='running query' timeout=60sec
if not result:
  // cached, or we lost the race.
  get data, loop if it's running query, or return if data
else:
  // not cached, we win the race
  run query
  replace cached flag with real value 
```

I'm not sure if this race condition is worth fixing though, as the worst case on a race condition is that we do an extra db query, and that will only happen for parallel users, not one user.